### PR TITLE
upgrade to polkadot v0.9.26

### PR DESCRIPTION
**Pull request by rajesh-nodle**
Fixes  https://github.com/NodleCode/chain/issues/631

**Completed**

- pallets
	* allocations :: Done
	* grants :: Done
	* mandate :: Done
	* poa :: Done
	* reserve :: Done
	* staking :: Done
* Runtime
        *  eden :: Done
* node :: Done

* Benchmarking on all Pallets :: Done

* Integration test :: Done

* Runtime or storage upgrade ( Node Client or Host needs to be latest v0.9.26 ).

* try-runtime :: Done (Trace prints not working - looks like substrate cli issue)

### CODE REVIEW

**Comment by rajesh-nodle** [comment]
Parachain runtime Upgrade status ...

**Cfg1**

- nodle-chain ver (v2.0.21) Host
- runtime wasm blob from v0.9.26 

Failed, looks like migrated v0.9.26 runtime is looking for Host 
function which is not part of (v2.0.21) host client.

```
2022-07-28 15:26:54 [Parachain] 🔖 Pre-sealed block for proposal at 117. Hash now 0xaeb69013dbbdcd713f20d4ce7db77cfb77cd03086e63fbe12f05e104a19cb622, previously 0x15c6c00fc049e160fc9c6ed0ea7c21076809c310eb8e2f9ba4c05be77a1c45ac.    
2022-07-28 15:26:54 [Parachain] ✨ Imported #117 (0xaeb6…b622)    
2022-07-28 15:26:54 [Parachain] PoV size { header: 0.1796875kb, extrinsics: 2.857421875kb, storage_proof: 1224.09765625kb }
2022-07-28 15:26:54 [Parachain] Compressed PoV size: 1227.177734375kb
2022-07-28 15:26:55 [Parachain] Cannot create a runtime error=Other("runtime requires function imports which are not present on the host: 'env:ext_benchmarking_read_write_count_version_1', 'env:ext_benchmarking_set_whitelist_version_1', 'env:ext_benchmarking_reset_read_write_count_version_1', 'env:ext_benchmarking_wipe_db_version_1', 'env:ext_benchmarking_add_to_whitelist_version_1', 'env:ext_benchmarking_get_read_and_written_keys_version_1', 'env:ext_benchmarking_current_time_version_1', 'env:ext_benchmarking_proof_size_version_1', 'env:ext_benchmarking_commit_db_version_1'")
2022-07-28 15:26:55 [Parachain] Failed to collect collation info. error=Application(VersionInvalid("runtime requires function imports which are not present on the host: 'env:ext_benchmarking_read_write_count_version_1', 'env:ext_benchmarking_set_whitelist_version_1', 'env:ext_benchmarking_reset_read_write_count_version_1', 'env:ext_benchmarking_wipe_db_version_1', 'env:ext_benchmarking_add_to_whitelist_version_1', 'env:ext_benchmarking_get_read_and_written_keys_version_1', 'env:ext_benchmarking_current_time_version_1', 'env:ext_benchmarking_proof_size_version_1', 'env:ext_benchmarking_commit_db_version_1'"))
2022-07-28 15:26:55 [Parachain] 💤 Idle (4 peers), best: #116 (0xf151…edd3), finalized #116 (0xf151…edd3), ⬇ 0.1kiB/s ⬆ 0.2kiB/s    
2022-07-28 15:26:55 [Relaychain] 💤 Idle (6 peers), best: #341 (0xae90…b415), finalized #338 (0xf1c7…e7f4), ⬇ 4.1kiB/s ⬆ 2.5kiB/s    
2022-07-28 15:27:00 [Relaychain] ✨ Imported #342 (0x160d…2af4)    
2022-07-28 15:27:00 [Parachain] Starting collation. relay_parent=0x160d2d3a82246abf263f0470d44b1a95d9965f3e5088f41c00ae46a7f08e2af4 at=0xf151cef5235351080175f68d4b416b245fb607f99cf435e74338032fca48edd3
2022-07-28 15:27:00 [Parachain] 💤 Idle (4 peers), best: #116 (0xf151…edd3), finalized #116 (0xf151…edd3), ⬇ 0 ⬆ 0    
2022-07-28 15:27:00 [Relaychain] 💤 Idle (6 peers), best: #342 (0x160d…2af4), finalized #339 (0xf4a6…03d7), ⬇ 1.8kiB/s ⬆ 2.0kiB/s    
```

**Cfg2**

- nodle-chain ver v0.9.26 Host
- runtime wasm blob from v0.9.26 

Test in progress ...

**Comment by simonsso** [comment]
Is everything committed to this branch? 

It did not build on my machine:
```
simson@tama:~/nodle/chain$ cargo build --release
    Updating git repository `https://github.com/paritytech/substrate`
   Compiling librocksdb-sys v0.6.1+6.28.2
   Compiling zstd-safe v5.0.2+zstd.1.5.2
   Compiling flate2 v1.0.24
   Compiling lz4 v1.23.3
   Compiling expander v0.0.4
   Compiling addr2line v0.17.0
   Compiling ss58-registry v1.25.0
   Compiling wasmtime-types v0.38.3
   Compiling wasmtime-jit-debug v0.38.3
   Compiling frame-support-procedural-tools v4.0.0-dev (https://github.com/paritytech/substrate?branch=polkadot-v0.9.20#563f4820)
   Compiling sp-api-proc-macro v4.0.0-dev (https://github.com/paritytech/substrate?branch=polkadot-v0.9.20#563f4820)
   Compiling cumulus-pallet-parachain-system-proc-macro v0.1.0 (https://github.com/paritytech/cumulus?branch=polkadot-v0.9.26#e43b8b87)
   Compiling cranelift-codegen v0.85.3
   Compiling prost-build v0.10.4
   Compiling snow v0.9.0
   Compiling trust-dns-resolver v0.21.2
error[E0603]: module `group` is private
   --> /home/simson/.cargo/git/checkouts/substrate-7e08433d4c370a21/563f482/frame/support/procedural/tools/src/syn_ext.rs:50:53
    |
50  |                 let syn::group::$name { token, content } = syn::group::$parse(input)?;
    |                                                                 ^^^^^ private module
    |
note: the module `group` is defined here
   --> /home/simson/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.98/src/lib.rs:304:1
    |
304 | mod group;
    | ^^^^^^^^^

error[E0603]: module `group` is private
   --> /home/simson/.cargo/git/checkouts/substrate-7e08433d4c370a21/563f482/frame/support/procedural/tools/src/syn_ext.rs:50:14
    |
50  |                 let syn::group::$name { token, content } = syn::group::$parse(input)?;
    |                          ^^^^^ private module
    |
note: the module `group` is defined here
   --> /home/simson/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.98/src/lib.rs:304:1
    |
304 | mod group;
    | ^^^^^^^^^

For more information about this error, try `rustc --explain E0603`.
   Compiling sp-core-hashing-proc-macro v5.0.0 (https://github.com/paritytech/substrate?branch=polkadot-v0.9.26#e8a7d161)
error: could not compile `frame-support-procedural-tools` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

**Comment by rajesh-nodle** [comment]
* Looks CI Build is OK, So no issues with integrity of commits,



* Good to verify the toolchain default, CI build uses `nightly` as default,
and I am using the same on my local dev machine.



* And as worst case option, better to make fresh checkout of the repo, and trigger fresh build.

**Comment by simonsso** [comment]
> 



> * Looks CI Build is OK, So no issues with integrity of commits,

I looked at CI status and saw only the red markers from code coverage.

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -120,7 +126,7 @@ fn eden_testnet_genesis(
 
 		// Consensus
 		validators_set: ValidatorsSetConfig {
-			members: collators.iter().map(|x| x.0.clone()).collect::<Vec<_>>(),
+			members: BoundedVec::try_from(validator_members).expect("Couldbe Max Overflow"),
```
Comment (simonsso) [inline node/src/chain_spec.rs:129]: expect is still an unwrap()

File: node/src/rpc.rs [hunk 2]
```diff
@@ -53,26 +53,22 @@ where
...
 	} = deps;
 
-	io.extend_with(SystemApi::to_delegate(FullSystem::new(
-		client.clone(),
-		pool,
-		deny_unsafe,
-	)));
-	io.extend_with(TransactionPaymentApi::to_delegate(TransactionPayment::new(client)));
+	module.merge(System::new(client.clone(), pool, deny_unsafe).into_rpc())?;
+	module.merge(TransactionPayment::new(client.clone()).into_rpc())?;
```
Comment (simonsso) [inline node/src/rpc.rs:71]:
```suggestion
	module.merge(TransactionPayment::new(client).into_rpc())?;
```
There is an automatic Clippy warning here:
GitHub Actions / clippy

redundant clone

warning: redundant clone
  --> node/src/rpc.rs:71:45
   |
71 |     module.merge(TransactionPayment::new(client.clone()).into_rpc())?;
   |                                                ^^^^^^^^ help: remove this
   |
   = note: `#[warn(clippy::redundant_clone)]` on by default
note: this value is dropped without further use
  --> node/src/rpc.rs:71:39
   |
71 |     module.merge(TransactionPayment::new(client.clone()).into_rpc())?;
   |                                          ^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone

File: node/src/chain_spec.rs (hunk 1 above)
Comment (rajesh-nodle) [reply node/src/chain_spec.rs:129]: Any alternate suggestion ?

Comment (simonsso) [inline node/src/chain_spec.rs:129]:
```suggestion
			members: BoundedVec::try_from(validator_members).unwrap_or_default(),
```

Yes I think a default approach is a more defensive alternative, but again this should never happen.

Comment (simonsso) [reply node/src/chain_spec.rs:129]: yes `unwrap_or_default()`

Comment (rajesh-nodle) [reply node/src/chain_spec.rs:129]: * In my view, this function is part of bootstrap sequence, so early detect of the error is good to make the corrective action early.

* Default is good suggestion but it leads to runtime error, may be landed up in no valid validatorset as part of POA. as a result may lead to no block production as latter part in the sequence.

* expect() or panic!() are not harm in this context, upstream template code is using it for checking the WASM binary.

Comment (simonsso) [reply node/src/chain_spec.rs:129]: I know I have seen it

**Pull request merged**